app/javascript/chat/chat.jsx

```diff
@@ -1028,7 +1029,7 @@ export default class Chat extends Component {
             >
               {'<'}
             </button>
-            <input placeholder="Filter" onKeyUp={this.triggerChannelFilter} />
+            <input placeholder="Filter" onKeyUp={debounce(this.triggerChannelFilter, 500)} />
```

Comment:
It feels weird to me to pass in a function wrapped in `debounce` as an argument to this `input`. I'd prefer if that was something that was encapsulated in the function itself; this could easily get lost and someone might not know that this function was being debounced. If we put this in a function, we could also add a comment to note why we're doing the debouncing here, specifically.

Reply:
I could see us doing something like this instead:

```
<input placeholder="Filter" onKeyUp={this.debouncedTriggerChannelFilter} />
```

where the `debouncedTriggerChannelFilter` could just do the work of encapsulating the debouncing logic.

Reply:
Can you show me how that would work in the React code? I tried for 2 hours getting it integrated with the function itself and I failed miserably which is how I ended up here. React components are not my strong suit.

Reply:
Hm, I played around with this a bit locally and couldn't get my imagined solution to work as expected. I feel like one _should_ be able to call a function defined on the component, `this`, from within another function defined on that some component, but maybe I'm making a mistake.

I also am having serious issues getting this stuff to actually work correctly (even on master, I'm running into a bunch of Elasticsearch `Elasticsearch::Transport::Transport::Errors::NotFound` errors, even though I definitely have some chat channels that are seeded, which is making this frustratingly hard to test on my own local branch. 🤷‍♀

In any case, I feel like you _should_ be able to do something like what we do for the search box keyup function [here](https://github.com/thepracticaldev/dev.to/blob/42771beb5d74e954ac5472585bf4e0db84467233/app/javascript/readingList/readingList.jsx#L42-L44). 

I basically have this in the chat component's `constructor`:
```
this.debouncedChannelFilter = debounce(this.triggerChannelFilter.bind(this), 500, {
  leading: true,
});
```

with this inside of the function that renders:
```
<input placeholder="Filter" onKeyUp={this.debouncedChannelFilter} />
```

Again, can't tell you if this is right because locally, ES just keeps erroring. 😞 It looks like @rhymes wrote the debounced search box functionality, maybe he knows what I'm doing wrong here and how to just wrap this in a function directly _on_ the component rather than binding it in the `constructor`?

Reply:
pretty sure I was missing a `this` somewhere in there bc I kept getting Reference not found errors but now I got it working. 🙏